src/main/java/org/ibp/api/java/impl/middleware/germplasm/GermplasmServiceImpl.java

```diff
@@ -371,21 +377,39 @@ public long countGermplasmMatches(final GermplasmMatchRequestDto germplasmMatchR
 		BaseValidator.checkNotNull(germplasmMatchRequestDto, "germplasm.match.request.null");
 		if (!germplasmMatchRequestDto.isValid()) {
 			this.errors = new MapBindingResult(new HashMap<String, String>(), GermplasmMatchRequestDto.class.getName());
-			errors.reject("germplasm.match.request.invalid", "");
+			this.errors.reject("germplasm.match.request.invalid", "");
 			throw new ApiRequestValidationException(this.errors.getAllErrors());
 		}
-		return germplasmService.countGermplasmMatches(germplasmMatchRequestDto);
+		return this.germplasmService.countGermplasmMatches(germplasmMatchRequestDto);
 	}
 
 	@Override
 	public List<GermplasmDto> findGermplasmMatches(final GermplasmMatchRequestDto germplasmMatchRequestDto, final Pageable pageable) {
 		BaseValidator.checkNotNull(germplasmMatchRequestDto, "germplasm.match.request.null");
 		if (!germplasmMatchRequestDto.isValid()) {
 			this.errors = new MapBindingResult(new HashMap<String, String>(), GermplasmMatchRequestDto.class.getName());
-			errors.reject("germplasm.match.request.invalid", "");
+			this.errors.reject("germplasm.match.request.invalid", "");
 			throw new ApiRequestValidationException(this.errors.getAllErrors());
 		}
-		return germplasmService.findGermplasmMatches(germplasmMatchRequestDto, pageable);
+		return this.germplasmService.findGermplasmMatches(germplasmMatchRequestDto, pageable);
+	}
+
+	@Override
+	public GermplasmDeleteResponse deleteGermplasm(final List<Integer> gids) {
+
+		this.errors = new MapBindingResult(new HashMap<>(), String.class.getName());
+		this.germplasmValidator.validateGids(this.errors, gids);
```

Comment:
If any of the GIDs are invalid or non-existing then the whole request will fail?

Reply:
If any of the GIDs are non-existing then the whole request will fail. But if all GIDs exist and some have validation error (has inventory, fixed group, etc..), the request will continue but only GIDs who passed the validation will be deleted.